### ISSUE TITLE
Config bugs

### ISSUE THREAD

**Issue by Warchant** [hyperledger/iroha#204]
This is how iroha reacts to absense of configs:
```
1488934134 ERROR (-A-) [config] Not found: /iroha/config/config.json
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted (core dumped)
1488934136 ERROR (-A-) [config] Not found: /iroha/config/sumeragi.json
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted (core dumped)
```

And, when I create empty config.json and sumeragi.json I can run iroha:
```
1488934263 DEBUG [config] load json is 
1488934263 ERROR (-A-) [config] Can't parse json: config/config.json
1488934263 DEBUG [config] load json is 
1488934263 ERROR (-A-) [config] Can't parse json: config/config.json
1488934263 DEBUG [config] load json is 
1488934263 ERROR (-A-) [config] Can't parse json: config/config.json
1488934263 DEBUG [config] load json is 
1488934263 ERROR (-A-) [config] Can't parse json: config/config.json
1488934263 INFO [main] process is :332
1488934263 DEBUG [config] load json is 
1488934263 ERROR (-A-) [config] Can't parse json: config/config.json
```

As a user I want to run iroha without `config.json` at all. I want to get only warning "there is no config, we will use default values" (something like this).

**Comment by motxx** [comment]
Thanks, I'm fixing now.

**Comment by motxx** [comment]
In iroha_config_with_json, config value returns as follows:
```
std::string IrohaConfigManager::getDatabasePath(const std::string& defaultValue) {
  return this->getParam<std::string>("database_path", defaultValue);
}
```

**Comment by luckychess** [comment]
@Warchant  My initial config implementation was exactly the same as you want it now but was rejected on code review.

**Comment by motxx** [comment]
@Warchant @luckychess Sorry for my late. I fixed at #206, but please tell me if something wrong or awful.
